### PR TITLE
feat(behavior_path_planner): output multiple candidate paths

### PR DESCRIPTION
**Pull request by rej55**
Signed-off-by: Fumiya Watanabe <rej55.g@gmail.com>

## Description

In the future, multiple modules in behavior_path_planner will be run.
To handle this, in this change, I change behavior_path_planner to output multiple candidate paths.

In this change, the name space of candidate path topics is also changed.

Before:
`/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/output/path_candidate`

After:
`/planning/path_candidate/**`

Furthermore, it also outputs multiple RTC status.
So, it becomes able to receive RTC commands if the modules request.

## Tests performed

Tested in the planning simulator.
See also https://github.com/autowarefoundation/autoware_launch/pull/125

- Lane change & Avoidance
Screenshot from 2022-12-11 22-08-36

- Pull out & Avoidance
Screenshot from 2022-12-11 22-13-53

- Pull over & Avoidance
Screenshot from 2022-12-11 22-22-45

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by rej55** [comment]
TIER IV internal test has been finished.
821/825 -> 823/825

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp [hunk 1]
```diff
@@ -95,6 +93,7 @@ class SceneModuleInterface
 
     const auto ns = std::string("~/debug/") + module_ns;
     pub_debug_marker_ = node.create_publisher<MarkerArray>(ns, 20);
+    pub_path_candidate_ = node.create_publisher<Path>("/planning/path_candidate/" + module_ns, 1);
```
Comment (TakaHoribe) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp:96]: please move publishers to behavior_path_planner_node in order to make pub/sub interface clear.

Comment (rej55) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/scene_module_interface.hpp]: done
https://github.com/autowarefoundation/autoware.universe/pull/2486/commits/e06088d31948e76df716dd0289d8d6da372ba285

**Review by TakaHoribe — APPROVED** [review]
LGTM!!

File: planning/behavior_path_planner/src/scene_module/scene_module_bt_node_interface.cpp [hunk 2]
```diff
@@ -68,7 +68,7 @@ BT::NodeStatus SceneModuleBTNodeInterface::tick()
       // std::exit(EXIT_FAILURE);  // TODO(Horibe) do appropriate handing
     }
     scene_module_->unlockRTCCommand();
-    return BT::NodeStatus::SUCCESS;
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/scene_module/scene_module_bt_node_interface.cpp:71]: @rej55 
what is this change for?

Comment (rej55) [reply planning/behavior_path_planner/src/scene_module/scene_module_bt_node_interface.cpp:71]: @kosuke55 
If we apply https://github.com/autowarefoundation/autoware.universe/pull/2442, the feature to output multiple execute requests is required.
To run fall back in the behavior tree, I changed the node status.

**Pull request merged**